src/plugins/intel_gpu/include/intel_gpu/runtime/half.hpp

```diff
@@ -28,6 +28,8 @@ class half_impl {
     explicit half_impl(T data, int /*direct_creation_tag*/) : _data(data) {}
 
     operator uint16_t() const { return _data; }
+    operator int32_t() const { return static_cast<int32_t>(_data); }
```

Comment:
Why do you cast raw representation of float16 data type to integer type?
I think it should be something like this:
`operator int32_t() const { return static_cast<int32_t>(half_to_float(_data)); }`

Reply:
Yes you are right, fixed